src/Orleans/Streams/PersistentStreams/PersistentStreamProducer.cs

```diff
@@ -50,7 +50,7 @@ public Task OnNextAsync(T item, StreamSequenceToken token)
             {
                 throw new ArgumentNullException("token", "Passing a non-null token to a non-rewindable IAsyncBatchObserver.");
             }
-            return queueAdapter.QueueMessageAsync(stream.StreamId.Guid, stream.StreamId.Namespace, item);
+            return queueAdapter.QueueMessageAsync(stream.StreamId.Guid, stream.StreamId.Namespace, item, Runtime.RequestContext.Export());
```

Comment:
We have an open issue regarding not passing the token into the adapter in these calls.  If we're going to incur the cost of a change to a public api we may as well make both changes at once, rather than having two breaking changes.